core/primitives/digest.hpp

```diff
@@ -58,24 +59,83 @@ namespace kagome::primitives {
   /// checked (yet).
   struct Consensus : public detail::DigestItemCommon {
-    /// Note: order of types in variant matters
-    boost::variant<Unused<0>,
-                   ScheduledChange,  // 1: (Auth C; N delay)
-                   ForcedChange,     // 2: (Auth C; N delay)
-                   OnDisabled,       // 3: Auth ID
-                   Pause,            // 4: N delay
-                   Resume>           // 5: N delay
-        payload = Unused<0>{};
+    using BabeDigest =
+        /// Note: order of types in variant matters
+        boost::variant<Unused<0>,
+                       NextEpochData,    // 1: (Auth C; R)
+                       OnDisabled,       // 2: Auth ID
+                       NextConfigData>;  // 3: c, S2nd
+
+    using GrandpaDigest =
+        /// Note: order of types in variant matters
+        boost::variant<Unused<0>,
+                       ScheduledChange,  // 1: (Auth C; N delay)
+                       ForcedChange,     // 2: (Auth C; N delay)
+                       OnDisabled,       // 3: Auth ID
+                       Pause,            // 4: N delay
+                       Resume>;          // 5: N delay
 
     Consensus() = default;
 
-    template <class A,
-              typename = std::enable_if_t<!std::is_same_v<A, Consensus>, void>>
-    Consensus(A &&a) : payload(std::forward<A>(a)){};
-
-    template <class A,
-              typename = std::enable_if_t<!std::is_same_v<A, Consensus>, void>>
-    Consensus(const A &a) : payload(a){};
+    // Note: this ctor is needed only for tests
+    template <class A>
+    Consensus(const A &a) {
+      if constexpr (std::is_same_v<A, NextEpochData>) {
+        consensus_engine_id = primitives::kBabeEngineId;
+        data = common::Buffer(scale::encode(BabeDigest(a)).value());
+      } else if constexpr (std::is_same_v<A, NextConfigData>) {
+        consensus_engine_id = primitives::kBabeEngineId;
+        data = common::Buffer(scale::encode(BabeDigest(a)).value());
+      } else if constexpr (std::is_same_v<A, ScheduledChange>) {
+        consensus_engine_id = primitives::kGrandpaEngineId;
+        data = common::Buffer(scale::encode(GrandpaDigest(a)).value());
+      } else if constexpr (std::is_same_v<A, ForcedChange>) {
+        consensus_engine_id = primitives::kGrandpaEngineId;
+        data = common::Buffer(scale::encode(GrandpaDigest(a)).value());
+      } else if constexpr (std::is_same_v<A, OnDisabled>) {
+        consensus_engine_id = primitives::kGrandpaEngineId;
+        data = common::Buffer(scale::encode(GrandpaDigest(a)).value());
+      } else if constexpr (std::is_same_v<A, Pause>) {
+        consensus_engine_id = primitives::kGrandpaEngineId;
+        data = common::Buffer(scale::encode(GrandpaDigest(a)).value());
+      } else if constexpr (std::is_same_v<A, Resume>) {
+        consensus_engine_id = primitives::kGrandpaEngineId;
+        data = common::Buffer(scale::encode(GrandpaDigest(a)).value());
```

Comment:
As far as I can tell bodies here are the same, maybe put all conditions in one ORed?

Reply:
Agree. It'll make code a bit cleaner.